From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VerteDinde <vertedinde@electronjs.org>
Date: Sun, 3 Apr 2022 21:23:00 -0700
Subject: Revert "mac: Make dump_syms invocation repeat "dump_syms" less often"

This reverts commit d655c91617185d365037730c4d9dc44f819980d1.

diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index b8abc10e48bdff0f4e6c3f8e1c4927bc6e0c2f79..cde6826da7af05d9220759c6c69a1335968366e6 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1296,15 +1296,16 @@ if (is_win) {
       outputs =
           [ "$root_out_dir/{{source_file_part}}-$chrome_version_full.breakpad" ]
 
-      dump_syms = "//third_party/breakpad:dump_syms($host_toolchain)"
-      args = rebase_path(outputs, root_build_dir) + [
-               rebase_path(get_label_info(dump_syms, "root_out_dir") + "/" +
-                               get_label_info(dump_syms, "name"),
-                           root_build_dir),
+      args = rebase_path(outputs, root_out_dir) + [
+               rebase_path(
+                   get_label_info(
+                           "//third_party/breakpad:dump_syms($host_toolchain)",
+                           "root_out_dir") + "/dump_syms",
+                   root_out_dir),
                "-g",
                rebase_path(
                    "$root_out_dir/{{source_file_part}}.dSYM/Contents/Resources/DWARF/{{source_file_part}}",
-                   root_build_dir),
+                   root_out_dir),
                "{{source}}",
              ]
 
@@ -1314,10 +1315,10 @@ if (is_win) {
         "//components/crash/core/app:chrome_crashpad_handler",
         "//third_party/angle:libEGL",
         "//third_party/angle:libGLESv2",
+        "//third_party/breakpad:dump_syms($host_toolchain)",
         "//third_party/swiftshader/src/OpenGL/libEGL:swiftshader_libEGL",
         "//third_party/swiftshader/src/OpenGL/libGLESv2:swiftshader_libGLESv2",
         "//third_party/swiftshader/src/Vulkan:swiftshader_libvulkan",
-        dump_syms,
       ]
       if (build_with_internal_optimization_guide) {
         deps += [ "//components/optimization_guide/internal:optimization_guide_internal" ]
diff --git a/chrome/updater/mac/BUILD.gn b/chrome/updater/mac/BUILD.gn
index 4ad1c9b95663b507dcc8be68403667eef31f57d0..f32cb2be87eb7db593903e540440de9a26f5f90b 100644
--- a/chrome/updater/mac/BUILD.gn
+++ b/chrome/updater/mac/BUILD.gn
@@ -349,22 +349,24 @@ if (!is_component_build && enable_dsyms) {
   action_foreach("syms") {
     script = "//build/redirect_stdout.py"
     sources = _symbols_sources
-    dump_syms = "//third_party/breakpad:dump_syms($host_toolchain)"
     public_deps = [
       ":updater_bundle",
-      dump_syms,
+      "//third_party/breakpad:dump_syms($host_toolchain)",
     ]
     outputs =
         [ "$root_out_dir/{{source_file_part}}-$chrome_version_full.breakpad" ]
 
-    args = rebase_path(outputs, root_build_dir) + [
-             rebase_path(get_label_info(dump_syms, "root_out_dir") + "/" +
-                             get_label_info(dump_syms, "name"),
-                         root_build_dir),
+    # Use an absolute path to dump_syms in case a user has it in their path.
+    args = rebase_path(outputs, root_out_dir) + [
+             rebase_path(
+                 get_label_info(
+                         "//third_party/breakpad:dump_syms($host_toolchain)",
+                         "root_out_dir") + "/dump_syms",
+                 root_out_dir),
              "-g",
              rebase_path(
                  "$root_out_dir/{{source_file_part}}.dSYM/Contents/Resources/DWARF/{{source_file_part}}",
-                 root_build_dir),
+                 root_out_dir),
              "{{source}}",
            ]
   }
